server/validate/email/validate.go

```diff
@@ -215,14 +215,25 @@ func (v *validator) Request(user t.Uid, email, lang, resp string, tmpToken []byt
 }
 
 // ResetSecret sends a message with instructions for resetting an authentication secret.
-func (v *validator) ResetSecret(email, scheme, lang string, tmpToken []byte) error {
+func (v *validator) ResetSecret(email, scheme, lang string, tmpToken []byte, params map[string]interface{}) error {
 	// Normalize email to make sure Unicode case collisions don't lead to security problems.
 	email = strings.ToLower(email)
 
 	token := make([]byte, base64.URLEncoding.EncodedLen(len(tmpToken)))
 	base64.URLEncoding.Encode(token, tmpToken)
 	body := new(bytes.Buffer)
+	var login string
```

Comment:
`login := "-"` would simplify logic

Reply:
Even better would be to leave login empty if missing an then use template condition to include login only if it's not missing, like
```
{{if .Login}} 
Login: {{.Login}} 
{{end}}
```

Reply:
even better:
```
{{with .Login}} 
Login: {{.}} 
{{end}}
```

Reply:
Thanks! Done.